RMG/3Day/k 번째 수.js

```diff
@@ -0,0 +1,22 @@
+/*
+ * array의 배열에서 [i, j, k]를 원소로 가진 2차원 배열 command가 매개변수로 주어질 때
+ * i부터 j까지 자른 후(인덱스값 아님) 오름차순으로 정리한 후 k번째 수를 뽑는 함수입니다.
+ * solution([1, 5, 2, 6, 3, 7, 4],[[2, 5, 3], [4, 4, 1], [1, 7, 3]] // [5, 6, 3]
+ */
+
+function solution(array, commands) {
+  let newArr = []; // slice후의 값을 담기 위한 빈 배열을 선언합니다.
+  let answerArr = []; // k번째의 수를 담기위한 빈 배열을 선언합니다.
+
+  for (let i = 0; i < commands.length; i++) {
+    // command수에 맞게 뽑아야하므로 .length를 이용하여 for문을 선언합니다.
+    let start = commands[i][0] - 1; // slice() 메소드의 동작 방식은 인덱스 값이 [(start < end)] = index[start] index[end-1] 가 됩니다.
+    let end = commands[i][1]; // 위에와 마찬가지로 [(start < end)] 이므로 그대로 지정해주었습니다.
+    let k = commands[i][2]; // k번째 수를 뽑기위하여 선언하였습니다.
+    newArr = array.slice(start, end); //배열의 slice()함수를 이용하였습니다.
+    newArr.sort((a, b) => a - b); // 자른 배열을 오름찰순으로 정리하였습니다.
+    answerArr.push(newArr[k - 1]); // 초기화가 되지 않고 누적이 되도록, push()를 이용하였습니다.
+  }
+
+  return answerArr;
+}
```

Comment:
```suggestion
function solution(array, commands) {
  let answerArr = [];

  for (let i = 0; i < commands.length; i++) {
    const start = commands[i][0] - 1;
    const end = commands[i][1];
    const k = commands[i][2];

    const sortedArray = array.slice(start, end).sort((a, b) => a - b);
    answerArr.push(sortedArray[k - 1]);
  }

  return answerArr;
}
```

각 commend 원소들을 변수에 알맞게 선언해줘서 코드를 이해하기 편하네요.
그런데 너무 많은 주석은 오히려 잘 짜여진 코드의 가독성을 해친다고 생각합니다.

주석으로 코드를 설명하는 것 보단 '코딩 컨벤션이 일정함', '적합한 변수명 사용' 등의 방법으로 어떻게 하면 코드를 읽기 좋게 만들 수 있을지 고민해보면 좋을 것 같습니다.

추가로 start, end, k 변수는 값이 변하지 않으므로 const 로 선언해주시는게 좋습니다. (값의 변경을 막기 위함.)

Reply:
단순히 코딩테스트라고만 생각하여 저의 풀이 방법을 적고 다시 볼 때 확실하게 기억하고 싶어서 하나하나 주석 처리했는데, 실전이라 생각하고 진행하겠습니다 !!

Reply:
> 단순히 코딩테스트라고만 생각하여 저의 풀이 방법을 적고 다시 볼 때 확실하게 기억하고 싶어서 하나하나 주석 처리했는데, 실전이라 생각하고 진행하겠습니다 !!

풀이 방법을 적고 싶으시다면 코드 위쪽에 따로 작성해주시는 것도 좋을 거 같아요.

Reply:
앞으로는 그렇게 진행하도록 하겠습니다 !!